internal/kms/vault_sa.go

```diff
@@ -303,7 +310,7 @@ func (kms *vaultTenantSA) getToken() (string, error) {
 		}
 	}
 
-	return "", fmt.Errorf("failed to find token in ServiceAccount %s/%s", kms.Tenant, kms.tenantSAName)
+	return kms.createToken(sa, c)
```

Comment:
scraping the auto-generated token for use outside a pod was never correct... is there a reason not to prefer the TokenRequest approach instead of falling back to it?

Reply:
> scraping the auto-generated token for use outside a pod was never correct... is there a reason not to prefer the TokenRequest approach instead of falling back to it?

The current approach so far used to fetch the token from the secret attached and createtoken was included to support k8s 1.24's way of token request api. 

We can probably only stick to token request api from following releases after making sure it stable with cephcsi?
(this patch will also be backported to 3.6)

cc @nixpanic